akka-actor/src/main/scala/akka/actor/FaultHandling.scala

```diff
@@ -106,53 +103,91 @@ trait SupervisorStrategyLowPriorityImplicits { this: SupervisorStrategy.type =>
 }
 
 object SupervisorStrategy extends SupervisorStrategyLowPriorityImplicits {
-  sealed trait Directive
+  sealed trait Directive {
+    private[akka] def logLevel: LogLevel
+  }
 
   /**
    * Resumes message processing for the failed Actor
    */
-  case object Resume extends Directive
+  case object Resume extends Resume(Logging.WarningLevel)
+
+  /**
+   * INTERNAL API
+   */
+  private[akka] class Resume(private[akka] val logLevel: LogLevel) extends Directive
```

Comment:
```suggestion
  @InternalApi
  private[akka] class Resume(private[akka] val logLevel: LogLevel) extends Directive
```

Could perhas be `sealed` as well?